### PR TITLE
Add scopes to the tasks defined by the sbt plugin

### LINKED ISSUE #692: Failing to compile nativelib should stop everything
Currently, failing the `nativelib` will just `println` an error message, but won't stop the whole process, which may still report success:

```
> tests/run
[info] Updating {file:/Users/martin/Documents/Projects.nosync/Duhemm/scala-native/}tests...
[info] Resolving jline#jline;2.14.3 ...
[info] Done updating.
[info] Compiling 5 Scala sources to /Users/martin/Documents/Projects.nosync/Duhemm/scala-native/unit-tests/target/scala-2.11/classes...
[warn] [0] /main/scala/scala/scalanative/IssuesSuite.scala:264: a pure expression does nothing in statement position; you may be omitting necessary parentheses
[warn] [1] /main/scala/scala/scalanative/IssuesSuite.scala:265: a pure expression does nothing in statement position; you may be omitting necessary parentheses
[warn] [2] unknown:0: there was one deprecation warning; re-run with -deprecation for details
[warn] /main/scala/scala/scalanative/IssuesSuite.scala: 264 [0], 265 [1]
[warn] unknown: 0 [2]
[error] /Users/martin/Documents/Projects.nosync/Duhemm/scala-native/unit-tests/target/scala-2.11/native/lib/statvfs.c:44:1: error: unknown type name 'foobar'
[error] foobar
[error] ^
[error] /Users/martin/Documents/Projects.nosync/Duhemm/scala-native/unit-tests/target/scala-2.11/native/lib/statvfs.c:46:1: error: expected identifier or '('
[error] int scalanative_fstatvfs(int fd, struct scalanative_statvfs *buf) {
[error] ^
[error] 2 errors generated.
Failed to compile native library runtime code.
[info] Linking (1635 ms)
[info] Discovered 2341 classes and 14739 methods
[info] Optimizing (8468 ms)
[info] Generating intermediate code (918 ms)
[info] Produced 53 files
[info] Compiling to native code (2914 ms)
[info] Linking native code (421 ms)
//
// Hundreds of lines
//
* tests.SuiteSuite$
  [ok] expects true
  [ok] does not expect false
  [ok] expects not false
  [ok] does not expect true
  [ok] expects A and throws A
  [ok] expects C and throws C with function
  [ok] expects A and throws B
  [ok] expects A and doesn't throw
[success] Total time: 17 s, completed May 3, 2017 9:51:53 AM
```

The error in `nativeCompileLib` is very unlikely to be noticed in this scenario.

### PR DESCRIPTION
**Pull request by Duhemm**
The changes to the scripted tests may be seen as a problem, but they are actually correct. The following build definition would fail at runtime, saying that there are undefined settings:

```scala
lazy val foo = taskKey[Unit]("...")

foo := {
  compile.value
  ()
}
```

Also, fixes #692

### CODE REVIEW

File: sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala [hunk 1]
```diff
@@ -235,18 +236,18 @@ object ScalaNativePluginInternal {
             logger.running(compilec)
             val result = Process(compilec, cwd) ! logger
             if (result != 0) {
-              println("Failed to compile native library runtime code.")
+              sys.error("Failed to compile native library runtime code.")
```
Comment (jonas) [inline sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala]: Looks like this is a fix of #692

Comment (Duhemm) [inline sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala]: Yup, I mentioned it in the commit message, but forgot to in the PR description. Thanks!

Comment (tindzk) [inline sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala]: Note that `sys.error` will be deprecated in 2.13: https://github.com/scala/scala/pull/5677. It's better to use `throw new RuntimeException` instead.

**Comment by densh** [comment]
The changes look good to me! 👍 
Please rebase on top of master and it's good to go!

**Comment by Duhemm** [comment]
Done!

**Comment by xuwei-k** [comment]
Maybe there is a regression or change specification of `sbt run`.
`sbt run` invoke **JVM main method** (not native) after this changes.

**Comment by densh** [comment]
@xuwei-k That's definitely a regression. 😱 /cc @Duhemm

**Comment by Duhemm** [comment]
@xuwei-k Do you have an example of project that shows this behavior? I can't reproduce it on the `sandbox` project in Scala Native's build, nor on another small project.

**Comment by xuwei-k** [comment]
> Do you have an example of project that shows this behavior? 

- https://github.com/xuwei-k/shapeless/commit/440339ccf0e2de98a422ba4d0e6cf
- https://travis-ci.org/xuwei-k/shapeless/jobs/244371413#L1445

```
[info] Compiling 50 Scala sources to /home/travis/build/xuwei-k/shapeless/examples/.native/target/scala-2.11/classes...
[info] Compiling 1 Scala source to /home/travis/build/xuwei-k/shapeless/nativeTest/target/scala-2.11/classes...
background log: info: Running shapeless.NativeMain 
background log: debug: Waiting for threads to exit or System.exit to be called.
background log: debug:   Classpath:
	/home/travis/build/xuwei-k/shapeless/nativeTest/target/scala-2.11/classes
	/home/travis/build/xuwei-k/shapeless/examples/.native/target/scala-2.11/classes
	/home/travis/build/xuwei-k/shapeless/core/native/target/scala-2.11/classes
	/home/travis/.ivy2/cache/org.scala-lang/scala-library/jars/scala-library-2.11.11.jar
	/home/travis/.ivy2/cache/org.scala-native/nativelib_native0.3_2.11/jars/nativelib_native0.3_2.11-0.3.0.jar
	/home/travis/.ivy2/cache/org.scala-lang/scala-reflect/jars/scala-reflect-2.11.11.jar
	/home/travis/.ivy2/cache/org.scala-native/javalib_native0.3_2.11/jars/javalib_native0.3_2.11-0.3.0.jar
	/home/travis/.ivy2/cache/org.scala-native/scalalib_native0.3_2.11/jars/scalalib_native0.3_2.11-0.3.0.jar
	/home/travis/.ivy2/cache/org.typelevel/macro-compat_2.11/jars/macro-compat_2.11-1.1.1.jar
	/home/travis/.ivy2/cache/org.scala-lang.modules/scala-parser-combinators_2.11/bundles/scala-parser-combinators_2.11-1.0.6.jar
background log: debug: Waiting for thread run-main-0 to terminate.
```

**Comment by densh** [comment]
@xuwei-k Can you please file an issue for this?

**Pull request merged**